### PR TITLE
Update dependencies in package.json

### LINKED ISSUE #178: Update package dependencies
e.g. node-sass to sass.

### PR DESCRIPTION
**Pull request by hangler**
Fixes #178. Removes a number of unused / unneeded packages, and updates a number of others. This should also address a number of security issues from outdated dependencies.

After a `yarn install`, `/ClientApp` should build (`yarn build`) with no warnings or errors.

In a few cases, I had to use `FixTypeLater` (an alias for `any`) for the sake of expediency when dealing with third-party libraries which do not have readily-available built-in typings. At some point we can try to use better typings.

### CODE REVIEW

**Comment by hangler** [comment]
Going to run an `npm audit` to see if there are any more packages that can be updated.

File: ClientApp/package.json [hunk 1]
```diff
@@ -4,66 +4,55 @@
...
+    "@types/redux": "^3.6.0",
     "bootstrap": "^4.4.1",
-    "class-transformer": "^0.3.1",
+    "class-transformer": "^0.5.1",
     "client-oauth2": "^4.2.5",
     "email-validator": "^2.0.4",
-    "eslint-config-prettier": "^6.7.0",
-    "eslint-config-react": "^1.1.7",
-    "eslint-plugin-prettier": "^3.1.1",
+    "gh-pages": "^4.0.0",
```
Comment (danebenal) [inline ClientApp/package.json:22]: Is this being used?

File: ClientApp/src/components/Login/CallbackHandler.tsx [hunk 2]
```diff
@@ -40,7 +41,7 @@ class CallbackPage extends React.Component<Props> {
             </p>
           </div>
         </div>
-      </CallbackComponent>
+      </>
```
Comment (danebenal) [inline ClientApp/src/components/Login/CallbackHandler.tsx:45]: What is this closing? The CallbackComponent is closing itself on line 32.

**Comment by hangler** [comment]
The only remaining vulnerability is this one, which [is almost certainly nothing to worry about, according to the Create React App team](https://github.com/facebook/create-react-app/issues/11174) ([see also Stack Overflow](https://stackoverflow.com/questions/71282206/github-dependabot-alert-inefficient-regular-expression-complexity-in-nth-check)):

```bash
% npm audit
# npm audit report

nth-check  <2.0.1
Severity: high
Inefficient Regular Expression Complexity in nth-check - https://github.com/advisories/GHSA-rp65-9cf3-cjxr
fix available via `npm audit fix --force`
Will install react-scripts@2.1.3, which is a breaking change
node_modules/svgo/node_modules/nth-check
  css-select  <=3.1.0
  Depends on vulnerable versions of nth-check
  node_modules/svgo/node_modules/css-select
    svgo  1.0.0 - 1.3.2
    Depends on vulnerable versions of css-select
    node_modules/svgo
      @svgr/plugin-svgo  <=5.5.0
      Depends on vulnerable versions of svgo
      node_modules/@svgr/plugin-svgo
        @svgr/webpack  4.0.0 - 5.5.0
        Depends on vulnerable versions of @svgr/plugin-svgo
        node_modules/@svgr/webpack
          react-scripts  >=2.1.4
          Depends on vulnerable versions of @svgr/webpack
          node_modules/react-scripts

6 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force
```

File: ClientApp/package.json (hunk 1 above)
Comment (hangler) [reply ClientApp/package.json:22]: Oops, no, good catch. Copied from a newly-generated Create React App template a bit too hastily!

**Review by danebenal — APPROVED** [review]

**Pull request merged**